src/core/utils/CustomMulticallJsonRpcProvider.ts

```diff
@@ -0,0 +1,31 @@
+import { providers } from 'ethers'
+import { MulticallQueue } from './multicall'
+
+export class CustomMulticallJsonRpcProvider extends providers.StaticJsonRpcProvider {
+  private multicallQueue: MulticallQueue
+  constructor({
+    url,
+    network,
+    multicallQueue,
+  }: {
+    url: string
+    network: providers.Networkish
+    multicallQueue: MulticallQueue
+  }) {
+    super(url, network)
+    this.multicallQueue = multicallQueue
+  }
+
+  async call(
+    transaction: providers.TransactionRequest,
+    blockTag?: providers.BlockTag | Promise<providers.BlockTag>,
+  ): Promise<string> {
+    console.log('Intercepted eth_call')
```

Comment:
I'd suggest `logger.debug` here, it might be a bit spammy otherwise.

Reply:
nice idea